src/renderer/renderer.ts

```diff
@@ -856,9 +857,35 @@ export class SDFGRenderer extends EventEmitter {
         else
             this.bgcolor = window.getComputedStyle(this.canvas).backgroundColor;
 
+
         // Create the initial SDFG layout
         // Loading animation already started in the file_read_complete function in sdfv.ts
         // to also include the JSON parsing step.
+        
+        // Initial layout fully collapsed
+        this.for_all_sdfg_elements(
+            (_t: SDFGElementGroup, _d: any, obj: any) => {
+                if ('is_collapsed' in obj.attributes &&
+                    !obj.type.endsWith('Exit'))
+                    obj.attributes.is_collapsed = true;
+            }
+        );
+        this.emit('collapse_state_changed', true, true);
+        this.relayout();
+
+        // Expand by one level
```

Comment:
I'm not sure about this mode for simple SDFGs, where we will start with states expanded but contents of maps collapsed. This might be good for huge graphs but some kernels will just require users to double click nodes, which some users do not know they can. @phschaad Thoughts?

Reply:
I could expand graphs depending on how large they are or add "expand" symbols to collapsed nodes. The main reasons for starting fully collapsed are the loading time when opening a file (54 seconds for the 255 MiB graph on my laptop) and readability of deeply nested graphs.